KFSPTS-12176 adding DVs to the create accounting document job

jhulslander opened the pull request:
This is ready for review.  Please note, in the unit tests, I added verifications for some variables in the sub objects of the Disbursement Voucher document.  I didn't add validations for each value as there a lot, and thought it was sufficient to just verify the sub objects had some values loaded.  If you think every value should be verified, please let me know, and will add the extra checks.

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherPreConferenceRegistrant.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    public void setPreConferenceRequestNuumber(String preConferenceRequestNuumber) {
+        this.preConferenceRequestNuumber = preConferenceRequestNuumber;
```
typo in method name, parameter and member variable preConferenceRequestNuumber

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherNonEmployeeTravel.java:
```diff
@@ -0,0 +1,331 @@
+import edu.cornell.kfs.sys.xmladapters.StringToJavaDateAdapter;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "NonEmoployeeTravel", namespace = StringUtils.EMPTY)
```
typo

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherNonEmployeeTravel.java:
```diff
@@ -0,0 +1,331 @@
+@XmlRootElement(name = "NonEmoployeeTravel", namespace = StringUtils.EMPTY)
+public class DisbursementVoucherNonEmployeeTravel {
+    
+    @XmlElement(name = "travler_name", namespace = StringUtils.EMPTY, required = false)
```
typo

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherNonEmployeeExpense.java:
```diff
@@ -0,0 +1,55 @@
+import edu.cornell.kfs.sys.xmladapters.KualiDecimalXmlAdapter;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "NonEmoployeeTravelExpense", namespace = StringUtils.EMPTY)
```
typo

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherNonEmployeeTravel.java:
```diff
@@ -0,0 +1,331 @@
+    @XmlElement(name = "personal_car_milage_amount", namespace = StringUtils.EMPTY, required = false)
+    @XmlJavaTypeAdapter(KualiDecimalXmlAdapter.class)
+    protected KualiDecimal personalCarMilageAmount;
+    
```
typos above (milage => mileage)

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherNonEmployeeTravel.java:
```diff
@@ -0,0 +1,331 @@
+    @XmlJavaTypeAdapter(KualiDecimalNullPossibleXmlAdapter.class)
+    protected KualiDecimal personalCarAmount;
+    
+    @XmlElementWrapper(name = "travler_expenses", namespace = StringUtils.EMPTY, required = true)
```
typo above (travler => traveler)?

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherNonEmployeeTravel.java:
```diff
@@ -0,0 +1,331 @@
+    public void setPersonalCarMilageAmount(KualiDecimal personalCarMilageAmount) {
+        this.personalCarMilageAmount = personalCarMilageAmount;
+    }
+
```
typos in the getter/setter (Milage => Mileage)

jhulslander commented on src/main/java/edu/cornell/kfs/fp/batch/service/impl/CuDisbursementVoucherDocumentGenerator.java:
```diff
@@ -0,0 +1,251 @@
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentAccountingLine;
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentEntry;
+import edu.cornell.kfs.fp.batch.xml.DisbursementVoucherDetail;
+import edu.cornell.kfs.fp.batch.xml.DisbursementVoucherNonEmployeeExpense;
```
I think adding xml to the end of these classes would make sense.  pre-pending the classes with CU would imply they are Cornell extension of the full Kuali business object, so I don't think we want to do that.  I will create a new user story to rename the preexisting classes edu.cornell.kfs.fp.batch.xml to include an XML at the end, where appropriate.

jhulslander commented:
Thanks for catching those errors,  I believe I have fixed all of the catches.  I also added additional unit tests to verify the names changes to the XML marshaling objects didn't break anything.

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherPaymentInfomrationXml.java:
```diff
@@ -0,0 +1,334 @@
+import edu.cornell.kfs.sys.xmladapters.StringToJavaDateAdapter;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "PaymentInformation", namespace = StringUtils.EMPTY)
```
Should this be payment_information?

jhulslander replied:
no, that tag is on DisbursementVoucherDetailXml, where this attribute is used, this root element just needs to be unique.  I will go through the new ojbects, and check for consistency

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherPreConferenceRegistrantXml.java:
```diff
@@ -0,0 +1,67 @@
+    
+    @XmlElement(name = "pre_conference_request_number", namespace = StringUtils.EMPTY, required = false)
+    @XmlJavaTypeAdapter(NormalizedStringAdapter.class)
+    protected String preConferenceRequestNuumber;
```
Typo Nuumber => Number. Also, the Getter and setter names/variables can be updated.

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherPreConferenceRegistrantXml.java:
```diff
@@ -0,0 +1,67 @@
+import edu.cornell.kfs.sys.xmladapters.KualiDecimalXmlAdapter;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "DisbursementVoucherPreConferenceRegistrant", namespace = StringUtils.EMPTY)
```
Should the name be registrant to match the documentation?

jhulslander replied:
The annotations of prepaid_registrants and registrant on DisbursementVoucherPrePaidTravelOverviewXml do match the documentation as fas as I can tell, the root element here just needs to be a unique identifier

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherDetailXml.java:
```diff
@@ -0,0 +1,109 @@
+import org.apache.commons.lang3.StringUtils;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "DisbursementVoucherDetail", namespace = StringUtils.EMPTY)
```
Does this correspond to dv_detail on the documentation?

jhulslander replied:
Yes, and the dv_detail is in the annotation on AccountingXmlDocumentEntry

ajd299 commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherPrePaidTravelOverviewXml.java:
```diff
@@ -0,0 +1,87 @@
+import edu.cornell.kfs.sys.xmladapters.StringToJavaDateAdapter;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "DisbursementVoucherPrePaidTravelOverview", namespace = StringUtils.EMPTY)
```
Should this be pre_paid_travel?

jhulslander replied:
nope, this just needs to be unique, pre_paid_travel is annotated in DisbursementVoucherDetailXml

ajd299 commented on src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java:
```diff
@@ -498,6 +508,35 @@ private void assertNoteIsCorrect(Note expectedNote, Note actualNote) {
+                actualDvDocument.getDvNonEmployeeTravel().getDisbVchrPerdiemRate());
+        assertEquals("conference destination not correct", expectedDvDocument.getDvPreConferenceDetail().getDvConferenceDestinationName(),
+                actualDvDocument.getDvPreConferenceDetail().getDvConferenceDestinationName());
+        assertEquals("non employee travler name not correct", expectedDvDocument.getDvNonEmployeeTravel().getDisbVchrNonEmpTravelerName(), 
```
Typo travler => traveler

ajd299 commented on src/test/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImplTest.java:
```diff
@@ -498,6 +508,35 @@ private void assertNoteIsCorrect(Note expectedNote, Note actualNote) {
+    
+    private void assertTravelExpenseCorrect(DisbursementVoucherNonEmployeeExpense expectedExpense, DisbursementVoucherNonEmployeeExpense actualExpense) {
+        assertEquals("expense code not correct", expectedExpense.getDisbVchrExpenseCode(), actualExpense.getDisbVchrExpenseCode());
+        assertEquals("expense compnay name not correct", expectedExpense.getDisbVchrExpenseCompanyName(), actualExpense.getDisbVchrExpenseCompanyName());
```
typo compnay => company

ajd299 commented on src/test/java/edu/cornell/kfs/fp/batch/xml/fixture/CuDisbursementVoucherDocumentFixture.java:
```diff
@@ -0,0 +1,59 @@
+    public final Integer nonEmployeeCarMileage;
+    public final List<CuDisbursementVoucherDocumentNonEmployeeTravelExpenseFixture> nonEmployeeTrevelerExpense;
+    public final List<CuDisbursementVoucherDocumentNonEmployeeTravelExpenseFixture> nonEmployeeTrevelerPrepaidExpense;
+    
```
The 2 variable names have a misspelling of Traveler (Treveler)

ajd299 commented on src/test/java/edu/cornell/kfs/fp/batch/xml/fixture/CuDisbursementVoucherDocumentFixture.java:
```diff
@@ -0,0 +1,59 @@
+    
+    private CuDisbursementVoucherDocumentFixture(String bankCode, String contactName, String paymentReasonCode, String payeeTypeCode, double perdiemRate, String conferenceDestination, 
+            String nonEmployeeTravelerName, Integer nonEmployeeCarMileage, CuDisbursementVoucherDocumentNonEmployeeTravelExpenseFixture[] nonEmployeeTrevelerExpenseArray,
+            CuDisbursementVoucherDocumentNonEmployeeTravelExpenseFixture[] nonEmployeeTrevelerPrepaidExpenseArray) {
```
these 2 parameter names also have a misspelling (Treveler => Traveler)

ajd299 commented on src/test/java/edu/cornell/kfs/fp/batch/xml/fixture/CuDisbursementVoucherDocumentNonEmployeeTravelExpenseFixture.java:
```diff
@@ -0,0 +1,21 @@
+    
+    
+    public final String expenseType;
+    public final String compnayName;
```
typo here compnay => company

jhulslander commented on src/main/java/edu/cornell/kfs/fp/batch/xml/DisbursementVoucherNonEmployeeExpenseXml.java:
```diff
@@ -0,0 +1,55 @@
+import edu.cornell.kfs.sys.xmladapters.KualiDecimalXmlAdapter;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "NonEmployeeTravelExpense", namespace = StringUtils.EMPTY)
```
This corresponds to the expenses in traveler_expenses and prepaid_expenses, the root element annotation  in this set up doesn't need to in the XML being marshalled, just a uniquer annotation identifier for  jaxB

ajd299 replied:
okay thanks for the explanation, please disregard my similar comments

jhulslander commented:
Thanks for noticing the root element tags, in the other XML objects, they have the same value as the annotation in where they are used, so to be consistent, I have changed these as well.

jhulslander commented:
OK, local validation  looks good with the latest changes, this ready for review agian

jhulslander commented:
I am going to close this PR, there a lot of comments that been fixed, I also just noticed some error handing isn't helpful
